packages/vscode-extension/src/pluginDebugger/copilotDebugLogOutput.ts

```diff
@@ -124,13 +190,134 @@ export class CopilotDebugLog {
+        );
+        if (Object.keys(capability.scopes).length > 0) {
+          Object.entries(capability.scopes).forEach(([key, value]) => {
+            debugConsole.appendLine(`       ${ANSIColors.WHITE}${key} - ${JSON.stringify(value)}`);
+          });
+        }
+      }
+    });
+  }
+
+  private logCapabilitiesWithExecutions(
+    debugConsole: vscode.DebugConsole,
+    capability: EnabledCapability
+  ): void {
+    debugConsole.append(
+      `${ANSIColors.GREEN}(√) ${ANSIColors.WHITE}Enabled capabilities: ${ANSIColors.MAGENTA}${capability.capabilityName}`
+    );
+    const capabilitiesOfThisTpe = this.capabilitiesDeveloperInfo?.capabilityExecutions?.filter(
+      (execution) => execution.name === capability.capabilityName
+    );
+
+    const capabilitiesContainAdditionalInfo = capabilitiesOfThisTpe?.some(
+      (execution) => execution.additionalDebugInfo
+    );
+
+    const logFileName = `Copilot-debug-${new Date().toISOString().replace(/-|:|\.\d+Z$/g, "")}.txt`;
+    const logFilePath = `${defaultExtensionLogPath}/${logFileName}`;
+
+    if (capabilitiesContainAdditionalInfo) {
+      debugConsole.appendLine("");
+      capabilitiesOfThisTpe?.forEach((execution, index) => {
+        const logFileName = `Copilot-debug-${new Date()
+          .toISOString()
+          .replace(/-|:|\.\d+Z$/g, "")}-${index}.txt`;
+        const logFilePath = `${defaultExtensionLogPath}/${logFileName}`;
+        writeExecutionDetailsToFile(logFilePath, JSON.stringify(execution, null, 2));
+        debugConsole.appendLine(
+          `       ${ANSIColors.WHITE} ${execution.name} ${index} ${
+            ANSIColors.WHITE
+          }• ${CopilotDebugLog.getExecutionStatusColor(
+            execution.status
+          )} Execution status ${CopilotDebugLog.getExecutionStatusText(execution.status)} ${
+            ANSIColors.WHITE
+          }refer to ${ANSIColors.BLUE}${logFilePath}${ANSIColors.WHITE} for all details.` // Replace execution.name with actual name
+        );
+      });
+    } else {
+      const finalExecution = capabilitiesOfThisTpe?.[capabilitiesOfThisTpe.length - 1];
+      writeExecutionDetailsToFile(logFilePath, JSON.stringify(finalExecution, null, 2));
+      debugConsole.appendLine(
+        ` ${ANSIColors.WHITE}• ${CopilotDebugLog.getExecutionStatusColor(
+          finalExecution?.status ?? ""
+        )} Execution status: ${CopilotDebugLog.getExecutionStatusText(
+          finalExecution?.status ?? ""
+        )}${ANSIColors.WHITE}, refer to ${ANSIColors.BLUE}${logFilePath}${
+          ANSIColors.WHITE
+        } for all details.`
+      );
+    }
+  }
+
+  private static getExecutionStatusColor(executionStatus: string): ANSIColors {
+    if (executionStatus.toString() === "1") {
+      return ANSIColors.GREEN;
+    } else {
+      return ANSIColors.RED;
+    }
+  }
+
+  private static getExecutionStatusText(executionStatus: string): string {
+    if (executionStatus.toString() === "1") {
```

Comment:
`executionStatus` is already a string

```suggestion
    if (executionStatus === "1") {
```